File: java/src/main/java/com/example/backend/application/lock/LockService.java
package com.example.backend.application.lock;

/**
 * 锁 Service 接口
 */
public interface LockService {

    /**
     * 获取一个锁，返回 AutoCloseable 对象，使用 try-with-resources 语法自动释放锁
     */
    AutoCloseable lock(String key);
}

File: backend-example-dotnet/src/application/lock/LockService.cs
namespace BackendExample.Application
{
    /// <summary>
    /// 锁 Service 接口
    /// </summary>
    public interface LockService
    {
        /// <summary>
        /// 同步加锁
        /// </summary>
        IDisposable Lock(string key, int timeout = 10);

        /// <summary>
        /// 异步加锁
        /// </summary>
        ValueTask<IAsyncDisposable> LockAsync(string key, int timeout = 10);
    }
}
